kotlinx-coroutines-core/jvm/src/channels/Channels.kt

```diff
@@ -10,12 +10,25 @@ package kotlinx.coroutines.channels
 import kotlinx.coroutines.*
 
 /**
- * Adds [element] into to this channel, **blocking** the caller while this channel is full,
- * or throws exception if the channel [Channel.isClosedForSend] (see [Channel.close] for details).
+ * ### Deprecation note.
  *
- * This is a way to call [Channel.send] method inside a blocking code using [runBlocking],
- * so this function should not be used from coroutine.
+ * This method is deprecated in the favour of [trySendBlocking].
+ *
+ * `sendBlocking` is considered to be dangerous primitive -- it throws
+ * if the channel was closed or, more commonly, cancelled.
+ * Cancellation exceptions are not ignored by non-blocking code and frequently
+ * trigger internal failures.
+ *
+ * These bugs were hard-to-spot during code review and also forced users to write
+ * their own wrappers around `sendBlocking`, so it was decided to deprecate
+ * this function and provide a more explicit primitive instead.
```

Comment:
It would be great to add an example of broken code here, too.

Reply:
Good suggested, added one from the recently reported bug